src/ElvTenant.js

```diff
@@ -802,6 +802,106 @@ class ElvTenant {
 
     return res;
   }
+
+  async TenantFundUser({ asUrl, tenantId, userAddress, amount = 0.2}) {
+
+    console.log("TenantFundUser");
+    console.log(`as_url: ${asUrl}`);
+    console.log(`tenant_id: ${tenantId}`);
+    console.log(`user_address: ${userAddress}`);
+    console.log(`amount: ${amount}`);
+
+    // Initialize configuration
+    const config = {
+      configUrl: Config.networks[Config.net],
+      mainObjectId: Config.mainObjects[Config.net],
+    };
+
+    // Initialize EluvioLive and ElvAccount
+    const eluvioLive = new EluvioLive(config);
+    await eluvioLive.Init({
+      debugLogging: this.debug,
+      asUrl
+    });
+
+    const elvAccount = new ElvAccount({
+      configUrl: Config.networks[Config.net],
+      debugLogging: this.debug,
+    });
+    await elvAccount.Init({ privateKey: process.env.PRIVATE_KEY });
+
+    // check valid user address
+    if (!Utils.ValidAddress(userAddress)) {
+      throw Error(`Invalid user address provided: ${userAddress}`);
+    }
+    const usrAddr = Utils.FormatAddress(userAddress);
+
+    // check tenant root key
+    const tenantAddr = Utils.HashToAddress(tenantId);
+    if (!Utils.ValidAddress(tenantAddr)) {
+      throw Error(`Invalid tenantId provided: ${tenantId}`);
+    }
+    const abi = fs.readFileSync(
+      path.resolve(__dirname, "../contracts/v3/BaseTenantSpace.abi")
+    );
+    const signerAddr = elvAccount.client.signer.address;
+    let isAdmin = await elvAccount.client.CallContractMethod({
```

Comment:
i bet encapsulating this will kill lots of duplicated lines of code, i count like 8 in this file alone...  would be awesome change :)

Reply:
I will make a new PR for this change, lot of files needs to cleaned.